tmt/base.py

```diff
@@ -2130,6 +2130,21 @@ def lint_unknown_keys(self) -> LinterReturn:
 
         yield LinterOutcome.PASS, 'correct keys are used'
 
+    def lint_empty_env_files(self) -> LinterReturn:
+        """ P001: env files are not empty """
+
+        env_files = self.node.get("environment-file") or []
+        if env_files:
```

Comment:
Is the condition needed? What about dropping it altogether? The `for` cycle would handle an empty `list` itself.

Reply:
It gives a better message: when the list is empty, you get an immediate `PASS`, while with a loop it would be hard to report such a pass as the loop would not indicate the number of iterations it took.

Reply:
Not sure I understand. If the list is empty, the `for` loop would be skipped and you get right to the `pass` statement. But yes, if the `return` statement is misplaced, then it's a completely different story.

Reply:
Yes, `return` is misplaced. I suppose it's structured as other lint checks, e.g.:

```python
    def lint_unknown_keys(self) -> LinterReturn:
        """ T001: all keys are known """

        # We don't want adjust in show/export so it is not yet in Test._keys
        invalid_keys = self._lint_keys(EXTRA_TEST_KEYS + OBSOLETED_TEST_KEYS + ['adjust'])

        if invalid_keys:
            for key in invalid_keys:
                yield LinterOutcome.FAIL, f'unknown key "{key}" is used'

            return

        yield LinterOutcome.PASS, 'correct keys are used'
```

Without the `if invalid_keys`, code would run the loop, finish the loop, and continue to `yield PASS`, which would be fine when `invalid_keys` is empty, but incorrect when it's not empty. Hence the test & hard `return` in that branch.

Reply:
Updated !

Reply:
The `PASS` statement, as suggested above, is missing. Thus you would never get ok:

```
> tmt lint
/plans/example
pass C000 fmf node passes schema validation
pass C001 summary key is set and is reasonably long
pass P001 correct keys are used
pass P002 execute step defined with "how"
pass P003 execute step methods are all known
skip P004 discover step is not defined
skip P005 no remote fmf ids defined
pass P006 phases have unique names
pass P007 execute phase 'default-0' does not require specific guest
```

Reply:
Updated !